clients/hadoopfs/src/test/java/io/lakefs/contract/TestLakeFSFileSystemContract.java

```diff
@@ -51,7 +47,15 @@ protected Path path(String pathString) {
   }
 
   public void testRenameFileAsExistingFile() throws Exception {
-    intercept(FileAlreadyExistsException.class, super::testRenameFileAsExistingFile);
+    if (!renameSupported()) return;
+
+    Path src = path("/test/hadoop/file");
+    createFile(src);
+    Path dst = path("/test/new/newfile");
+    createFile(dst);
+    // s3 doesn't support rename option
+    // rename-overwrites-dest is always allowed.
+    rename(src, dst, true, false, true);
```

Comment:
Can we verify the return value?

Reply:
not return value - this is not fs `rename` it is actual how the base contract tests rename.
what it does under the hood:
```go
    assertEquals("Rename result", renameSucceeded, fs.rename(src, dst));
    assertEquals("Source exists", srcExists, fs.exists(src));
    assertEquals("Destination exists" + dst, dstExists, fs.exists(dst));
```